Source/AviationWeather.Test/Source/MetarTests.cpp

```diff
@@ -321,82 +323,134 @@ TEST_CLASS(MetarTests)
+        Assert::AreEqual(aw::flight_category::mvfr, koak_metar.flight_category());
+        Assert::AreEqual(aw::flight_category::mvfr, ksfo_metar.flight_category());
 
         aw::metar::metar_info kcec_metar("KCEC 102356Z AUTO 26009KT 10SM OVC009 18/14 A2997 RMK AO2 SLP149 T01780144 10206 20161 50001");
         aw::metar::metar_info kacv_metar("KACV 102353Z AUTO 28007KT 10SM OVC009 18/16 A2997 RMK AO2 SLP152 T01830156 10217 20178 51001");
-        Assert::AreEqual(static_cast<uint8_t>(aw::flight_category::ifr), static_cast<uint8_t>(kcec_metar.flight_category()));
-        Assert::AreEqual(static_cast<uint8_t>(aw::flight_category::ifr), static_cast<uint8_t>(kacv_metar.flight_category()));
+        Assert::AreEqual(aw::flight_category::ifr, kcec_metar.flight_category());
+        Assert::AreEqual(aw::flight_category::ifr, kacv_metar.flight_category());
 
         aw::metar::metar_info konp_metar("KONP 102355Z AUTO 27003KT 1 1/2SM BR OVC003 16/14 A3000 RMK AO2");
         aw::metar::metar_info kcho_metar("KCHO 102353Z 16007KT 2SM BR OVC003 22/21 A2991 RMK AO2 RAE48 SLP122 P0002 60003 T02220211 10233 20222 56011");
-        Assert::AreEqual(static_cast<uint8_t>(aw::flight_category::lifr), static_cast<uint8_t>(konp_metar.flight_category()));
-        Assert::AreEqual(static_cast<uint8_t>(aw::flight_category::lifr), static_cast<uint8_t>(kcho_metar.flight_category()));
+        Assert::AreEqual(aw::flight_category::lifr, konp_metar.flight_category());
+        Assert::AreEqual(aw::flight_category::lifr, kcho_metar.flight_category());
+    }
+
+    TEST_METHOD(METAR_VariableWinds)
+    {
+        aw::metar::metar_info m1("KSFO 112056Z VRB17KT 10SM FEW010 SCT180 22/12 A2994 RMK AO2 SLP137 T02220122 58006");
+
+        Assert::IsTrue(m1.wind_group.is_variable());
+        Assert::AreEqual(UINT16_MAX, m1.wind_group.direction);
+        Assert::AreEqual(static_cast<uint8_t>(17), m1.wind_group.wind_speed);
+        Assert::AreEqual(static_cast<uint8_t>(0), m1.wind_group.gust_speed);
+        Assert::AreEqual(static_cast<uint8_t>(0), m1.wind_group.gust_factor());
+        Assert::AreEqual(UINT16_MAX, m1.wind_group.variation_lower);
+        Assert::AreEqual(UINT16_MAX, m1.wind_group.variation_upper);
+
+        aw::metar::metar_info m2("KRHV 101451Z 29014G18KT 10SM SKC 27/13 A2990");
+
+        Assert::IsFalse(m2.wind_group.is_variable());
+        Assert::AreEqual(static_cast<uint16_t>(290), m2.wind_group.direction);
+        Assert::AreEqual(static_cast<uint8_t>(14), m2.wind_group.wind_speed);
+        Assert::AreEqual(static_cast<uint8_t>(18), m2.wind_group.gust_speed);
+        Assert::AreEqual(static_cast<uint8_t>(4), m2.wind_group.gust_factor());
+        Assert::AreEqual(UINT16_MAX, m2.wind_group.variation_lower);
+        Assert::AreEqual(UINT16_MAX, m2.wind_group.variation_upper);
+
+        aw::metar::metar_info m3("KRHV 101850Z VRB14G18KT 10SM SKC 27/13 A2990");
```

Comment:
According to FMH-1 12.6.5 (b) VRB is only used for when winds are 6 knots or less, so technically this is an invalid METAR.